src/Vivait/TenantBundle/Locator/HeaderLocator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Vivait\TenantBundle\Locator;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * @see HostnameLocatorSpec
+ */
+class HeaderLocator implements TenantLocator
+{
+    private $request;
+
+    /**
+     * @var string
+     */
+    private $tenant_key;
+
+    public function __construct(Request $request, $tenant_key)
+    {
+        $this->request = $request;
+        $this->tenant_key = $tenant_key;
+    }
+
+    /**
+     * @return string Tenant key
+     */
+    public function getTenant()
+    {
+        if (!$this->request->headers->has($this->tenant_key)) {
+            throw new \RuntimeException('Could not locate a tenant in the header');
```

Comment:
Symfony should have registered its exception handler by this point, but I'll double check. Although if it hasn't, really it's a problem for the tenant bundle, not this driver.